weave/trace_server/file_storage.py

```diff
@@ -181,9 +208,13 @@ class AzureStorageClient(FileStorageClient):
     """Azure Blob Storage implementation supporting both connection string and account credentials."""
 
     def __init__(
-        self, credentials: Union[AzureConnectionCredentials, AzureAccountCredentials]
+        self,
+        base_uri: FileStorageURI,
+        credentials: Union[AzureConnectionCredentials, AzureAccountCredentials],
     ):
         """Initialize Azure client with either connection string or account credentials."""
+        assert isinstance(base_uri, AzureFileStorageURI)
```

Comment:
interesting, we prefer this over typing `base_uri`? or can't?

Reply:
It violates the `Liskov Substitution Principle`: "Method arguments being contravariant not only means that we can widen their types, but also that we cannot make their types more narrow." https://madewithlove.com/blog/liskov-substitution-principle-explained/

Reply:
gah silly Liskov! yeah